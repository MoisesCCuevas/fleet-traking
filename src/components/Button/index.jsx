
/**
 * Button component renders a customizable button element.
 *
 * @param {Object} props - The properties object.
 * @param {string} props.type - The type of the button (e.g., "button", "submit").
 * @param {function} props.onClick - The function to call when the button is clicked.
 * @param {React.ReactNode} props.children - The content to be displayed inside the button.
 * @param {boolean} [props.disabled=false] - Whether the button is disabled.
 * @returns {JSX.Element} The rendered button component.
 */
import React from "react"

export default function Button(props) {
  const { type, onClick, children, disabled } = props
  return (
    <button
      className="px-4 py-1 shadow-md bg-slate-600 rounded-md text-white disabled:opacity-55 flex items-center justify-center gap-2 text-nowrap"
      type={type}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </button>
  )
}
  