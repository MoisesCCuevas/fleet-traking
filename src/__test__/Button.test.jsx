import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Button from "@components/Button";

describe("Button component", () => {
  test("renders correctly with given props", () => {
    const { getByText } = render(
      <Button type="button" onClick={() => {}}>
        Click me
      </Button>
    );
    expect(getByText("Click me")).toMatchSnapshot();
  });

  test("calls onClick handler when clicked", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Button type="button" onClick={handleClick}>
        Click me
      </Button>
    );
    fireEvent.click(getByText("Click me"));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("is disabled when disabled prop is true", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Button type="button" onClick={() => {}} disabled={true}>
        Click me
      </Button>
    );
    fireEvent.click(getByText("Click me"));
    expect(handleClick).toHaveBeenCalledTimes(0);
  });
});
