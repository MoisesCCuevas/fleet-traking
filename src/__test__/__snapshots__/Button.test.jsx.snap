// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly with given props 1`] = `
<button
  class="px-4 py-1 shadow-md bg-slate-600 rounded-md text-white disabled:opacity-55 flex items-center justify-center gap-2 text-nowrap"
  type="button"
>
  Click me
</button>
`;
